executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Chain.hs

```diff
@@ -166,29 +235,56 @@ instance ( HasTypeReps hashAlgo
          ) => HasTrace (CHAIN hashAlgo) where
 
   envGen _traceLength
-    = Env <$> currentSlotGen
+    = Env <$> initialSlotGen
           <*> maxBlockSizeGen
-          <*> (transactionsEnvGen currentSlotGen)
+          <*> (transactionsEnvGen
+                initialSlotGen
+                closedVotingPeriodsGen
+              )
     where
-      currentSlotGen = Slot <$> Gen.integral (Range.constant 0 100)
+      initialSlotGen = Slot <$> Gen.integral (Range.constant 0 100)
+      -- TODO: Generate a realistic closed voting periods Map
+      closedVotingPeriodsGen = pure $ Map.empty
       -- For now we fix the maximum block size to an abstract size of 100
       maxBlockSizeGen = pure 100
       participantsGen = pure
                       $! Bimap.fromList
                       $  fmap (Core.vKey &&& Core.sKey)
                       $  fmap Core.keyPair
                       $  fmap Core.Owner $ [0 .. 10]
-      transactionsEnvGen gSlot
-        = Transaction.Env <$> gSlot
-                          <*> updatesEnvGen
-                          <*> (pure $ UTxO.Env)
-      updatesEnvGen = Update.Env <$> ideationEnvGen <*> implementationEnvGen
-      ideationEnvGen = participantsGen
-      implementationEnvGen = pure $ Implementation.Env
-
-  sigGen Env { maximumBlockSize, transactionsEnv } St { currentSlot, transactionsSt } =
+      transactionsEnvGen gSlot gClosedVotingPeriods
+        = Transaction.Env
```

Comment:
Minor thing: we might want to be consistent with the way we format these multiline applicative operations chaining. If we put the `=` sign on the next line, then we might want to write something like:

```haskell
      transactionsEnvGen gSlot gClosedVotingPeriods
        =   Transaction.Env
        <$> gSlot
        <*> gClosedVotingPeriods
        <*> updatesEnvGen gSlot gClosedVotingPeriods
        <*> (pure $ UTxO.Env)
```